lib/helpers.js

```diff
@@ -183,7 +183,7 @@ create buffer for block2 option
 module.exports.createBlock2 = function createBlock2(requestedBlock) {
   var byte
   var szx = Math.log(requestedBlock.size)/Math.log(2) - 4
-  var m = ((requestedBlock.moreBlock2==true)?0:1)
+  var m = ((requestedBlock.moreBlock2==true)?1:0)
```

Comment:
`requestedBlock.moreBlock2` is a boolean, maybe this statement can be abbreviated as `var m = requestedBlock.moreBlock2 ? 1:0`.

Reply:
I think we should simply do this change if relevant here or leave this PR out of the current fix version.
In the end I'm not sure that you can assume it is a boolean, but in fact the change you suggest also fix other "truthy" cases and is equivalent

Reply:
Leaving out the `== true` is exactly equivalent, imho. If `moreBlock2` is not boolean JS does its coercion magic as this is not a `===`. I just didn't change it because I wanted to touch the least amount of code necessary.

Reply:
@JcBernack I do understand your point, in fact requestedBlock in this sentence will be processed as `!!requestedBlock.moreBlock2`, so event though the value is not boolean, like `undefined` it would be still be converted to a boolean. But in this case, we can pr it first and submit another pr to optimize it.